integration-test/src/main/java/com/sequenceiq/it/cloudbreak/ImageCatalogTests.java

```diff
@@ -0,0 +1,209 @@
+                .withUrl("https://" + INVALID_IMAGECATALOG_URL), "an imagecatalog request with invalid url"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertNotEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-url");
+                }),  "check imagecatalog is not created with invalid url");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testCreateInvalidImageCatalogUrlNoProtocol() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "-url-np")
+                .withUrl(INVALID_IMAGECATALOG_URL), "an imagecatalog request with url no protocol"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertNotEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-url-np");
+                }),  "check imagecatalog is not created with url no protocol");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testCreateDeleteValidCreateAgain() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.isCreatedDeleted()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request then delete"
+        );
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "-delete-create")
+                .withUrl(INVALID_IMAGECATALOG_URL), "same imagecatalog request again"
+        );
+        when(ImageCatalog.post(), "post the imagecatalog request");
+        then(ImageCatalog.assertThis(
+                (imageCatalog, t) -> {
+                    Assert.assertEquals(imageCatalog.getResponse().getName(), VALID_IMAGECATALOG_NAME + "-delete-create");
+                }),  "check imagecatalog is created when name was used and deleted before");
+    }
+
+    @Test(expectedExceptions = BadRequestException.class)
+    public void testDeleteCbDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(DEFAULT_IMAGECATALOG_NAME)
+        );
+        when(ImageCatalog.delete(), "try to delete cb default image catalog");
+    }
+
+    @Test
+    public void testSetNewDeleteDefaultImageCatalog() throws Exception {
+        given(CloudbreakClient.isCreated());
+        given(ImageCatalog.request()
+                .withName(VALID_IMAGECATALOG_NAME + "default")
+                .withUrl(VALID_IMAGECATALOG_URL), "an imagecatalog request and set as default"
+
+        );
+        when(ImageCatalog.post());
+        when(ImageCatalog.setDefault());
```

Comment:
I guess it should be two tests:
1) setdefault was called - it became default 
2) if there is a default imagecatalog - it cannot be deleted

Reply:
and may be a third one:
 3. given a default [x] ; when set new [y] as a default; then the [x] is not default anymore